File: analysisdata.py
import xlrd
import numpy as np
import pylab as pl

realdatafname = 'realpoint.xlsx'
trackingdatafname = 'trackingpoint.xlsx'
realdatabk = xlrd.open_workbook(realdatafname)  # 获取xls文件
trackingdatabk = xlrd.open_workbook(trackingdatafname)  # 获取xls文件
realdatash = realdatabk.sheet_by_name('Sheet1')  #通过名称获取bk中的sheet
trackingdatash = trackingdatabk.sheet_by_name('Sheet1')  #通过名称获取bk中的sheet

# nrows = realdatash.nrows  # 获取行数
# ncols = realdatash.ncols  # 获取列数
# print('nrows %d, ncols %d' % (nrows,ncols))
# cell_value = realdatash.cell_value(1, 1)  # 获取单元格（1，1）的值
# print(cell_value)

realdatax = np.array(realdatash.col_values(0))  # 获取第0列值的列表,并将其转换为array类型
realdatay = np.array(realdatash.col_values(1))
trackingdatax = np.array(trackingdatash.col_values(0))  # 获取第0列值的列表,并将其转换为array类型
trackingdatay = np.array(trackingdatash.col_values(1))
# print(realdatax)
errorx = np.absolute(trackingdatax - realdatax)
errory = np.absolute(trackingdatay - realdatay)
errordistant = np.sqrt(errorx * errorx + errory * errory)
pl.figure(1)
pl.xlabel('frames')
pl.ylabel('xerror(pixel)')
pl.title('The error of x direction')
pl.plot(errorx)  # use pylab to plot x and y
pl.figure(2)
pl.xlabel('frames')
pl.ylabel('yerror(pixel)')
pl.title('The error of y direction')
pl.plot(errory)  # use pylab to plot x and y
pl.figure(3)
pl.xlabel('frames')
pl.ylabel('xerror(pixel)')
pl.title('The error of x direction')
pl.plot(errordistant)  # use pylab to plot x and y
pl.show()   # show the plot on the screen

File: analysisdata.m
%本文件用于计算图像追踪中真实点与跟踪点之间的误差
%注：误差没有转换为实际的距离
realdata = xlsread('realpoint.xlsx');
trackingdata = xlsread('trackingpoint.xlsx');
realdatax = realdata(:, 1);
realdatay = realdata(:, 2);
trackingdatax = trackingdata(:, 1);
trackingdatay = trackingdata(:, 2);
xerror = abs(trackingdatax - realdatax);
yerror = abs(trackingdatay - realdatay);
distanterror = sqrt(xerror.^2 + yerror.^2);
figure('numbertitle','off','name','The error of x direction','color','white');
plot(xerror);title('The error of x direction');xlabel('frames');ylabel('error（pixel）');
figure('numbertitle','off','name','The error of y direction','color','white');
plot(yerror);title('The error of y direction');xlabel('frames');ylabel('error（pixel）');
figure('numbertitle','off','name','The error of relative distance','color','white');
plot(distanterror);title('The error of relative distance');xlabel('frames');ylabel('error（pixel）');
